### ISSUE TITLE
Got string ClaimType for JSON ClaimType

### ISSUE THREAD

**Issue by taufikdev88** [DuendeSoftware/foss#63]
**Which version of Duende IdentityServer are you using?**

Duende.IdentityModel version 7.0.8

**Which version of .NET are you using?**

9.0

**Describe the bug**

A JSON ClaimType identified as string ClaimType

**To Reproduce**

1. Request Access Token from Keycloak with claim with type "resource_access" or "realm_access"

**Expected behavior**

"resource_access" or "realm_access" claim with "JSON" ClaimType

**Log output/exception with stacktrace**

```
![Image](https://github.com/user-attachments/assets/77587216-c5b2-48d5-abd4-2e7512eda1d9)
```

**Additional context**

-

**Comment by AndersAbel** [comment]
Why is the `ClaimValueType` important? How is it used? Why do you want to set it to ``"JSON"`? That's not a constant value in the .NET `ClaimValueTypes`.

**Comment by taufikdev88** [comment]
I see how Microsoft create claim with "JSON" ClaimValueTypes from Microsoft.IdentityModel.Tokens.Jwt package 
[JwtPayload.cs](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs#L720)
and from debugged code 
![debugging result](https://github.com/user-attachments/assets/54e0171c-ade7-410b-83e2-2a1138ddf8f0)

I found that microsoft created 2 files with same content for JsonClaimValueTypes
[System.IdentityModel.Tokens.Jwt](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/System.IdentityModel.Tokens.Jwt/JsonClaimValueTypes.cs)
[Microsoft.IdentityModel.JsonWebTokens](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Microsoft.IdentityModel.JsonWebTokens/JsonClaimValueTypes.cs)

How do you prefer? How about create JsonClaimValueTypes inside the foss project?

I found interesting open-source project out there are trying to get claims from ClaimPrincipal with ClaimValueTypes specific to "JSON", where we can see in this project
[AuthService](https://github.com/NikiforovAll/keycloak-authorization-services-dotnet/blob/main/src/Keycloak.AuthServices.Common/Claims/KeycloakResourceClaimsExtensions.cs#L65)

in some case developer need to combine multiple authentication scheme like I did, I use very great project from duende "IdentityModel.AspNetCore.OAuth2Introspection" which is now the project rebranded to Duende.IdentityModel

I think when all projects use standard definition for ClaimValueTypes, it will be great

**Comment by AndersAbel** [comment]
Yes, the `System.IdentityModel.Tokens.Jwt` package introduced the `"JSON"` `ClaimValueType`. They are moving that functionality to the `Microsoft.IdentityModel.JsonWebTokens` namespace, so even if there are two files they are essentially the same.

Using the `"JSON"` `ClaimValueType` is an extension. We are considering if we should do it too, but we would need to know *why* it is important to you. Do you have any code/functionality that relies on that value and treats the claim values differently if they are marked as `"JSON"` vs `ClaimValueTypes.String`?

**Comment by taufikdev88** [comment]
I have to ensure the claim type is JSON ![Image](https://github.com/user-attachments/assets/1ede89d5-93da-4c7d-88ce-b19507c90541)

**Comment by AndersAbel** [comment]
How is that code called? Do you convert the claim value string to something else based on the `ClaimValueType`?

**Comment by taufikdev88** [comment]
![Image](https://github.com/user-attachments/assets/9aad4c23-aa06-463f-9edd-734f410b0c86)

**Comment by AndersAbel** [comment]
I still don't understand how you use the `ClaimValueType`. Why does it matter to you? What functionality is dependent on the value being set to "JSON"?
